### PR TITLE
Explicitly describe how selector rules are applied

### PR DESCRIPTION
**Pull request by gparrello**
## Description
Expand documentation on Resource Group selector rules. Behaviour of selector rules is implied by the example provided in the readme, yet it can be explicitly mentioned for clarity.

## Additional context and related issues
[Related conversation](https://trinodb.slack.com/archives/C0305TQ05KL/p1708611021590289) from Trino Slack community.

## Release notes

(X) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

**Review by alelapi — APPROVED** [review]

**Review by hashhar — COMMENTED** [review]
This is technically accurate.

@mosabua please take a look.

File: docs/src/main/sphinx/admin/resource-groups.md [hunk 1]
```diff
@@ -187,6 +187,8 @@ documentation](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java
 
 - `group` (required): the group these queries will run in.
 
+All rules within a single selector are applied following an AND logic
```
Comment (mosabua) [inline docs/src/main/sphinx/admin/resource-groups.md:190]:
```suggestion
All rules within a single selector are combined using a logic `AND`, and therefore all rules must match for a selector to be applied.
```

File: docs/src/main/sphinx/admin/resource-groups.md [hunk 2]
```diff
@@ -187,6 +187,8 @@ documentation](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java
...
+(i.e. all rules must match for a selector to be applied).
```
Comment (mosabua) [inline docs/src/main/sphinx/admin/resource-groups.md:191]:
```suggestion

```

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/admin/resource-groups.md (hunk 1 above)
Comment (hashhar) [reply docs/src/main/sphinx/admin/resource-groups.md:190]: `logic AND` --> `logical AND` I assume?

Comment (mosabua) [reply docs/src/main/sphinx/admin/resource-groups.md:190]: well.. either is fine grammatically from what I know .. maybe we can refer to the expert - @colebow  ?

**Comment by mosabua** [comment]
Please adjust to feedback and submit a CLA (if you did already thats fine .. it will be a bit until we process)

**Comment by mosabua** [comment]
@cla-bot check

**Comment by mosabua** [comment]
Replaced by https://github.com/trinodb/trino/pull/21165

**Comment by gparrello** [comment]
I apologize for the lack of response, need to review how I am receiving notifications from GitHub for this repo. Thanks for updating the docs in #21165 !

**Pull request closed without merging**